README.md

```diff
@@ -1,2 +1,139 @@
 
 [![Coverage status](https://codecov.io/gh/lionel-/coercer/branch/master/graph/badge.svg)](https://codecov.io/github/lionel-/coercer?branch=master)
+
+
+## Overview
+
+This package provides an experimental scaffholding and interface for
+principled automatic coercions in tidyverse and r-lib packages. These
+coercions will be particularly helpful for combining results together
+in split-apply-combine settings (dplyr operations on grouped tibbles,
+`purrr::pmap()`, ...).
+
+We first focus on features and extensibility but speed will also be
+addressed eventually with fast concatenation for base types.
+
+Install the development version from github with:
+
+```r
+# install.packages("devtools")
+devtools::install_github("lionel-/coercer")
+```
+
+
+## vec()
+
+`vec()` is a replacement for `base::c()` that uses a consistent and
+extensible mechanism for automatic coercions.
+
+In simple cases vec() works like c():
+
+```
+vec(1, 2L, TRUE)
+#> [1] 1 2 1
+
+c(1, 2L, TRUE)
+#> [1] 1 2 1
+```
+
+However it supports more types and is extensible via a binary dispatch
+mechanism. Factors are well supported by default:
+
+```
+foo <- factor("foo")
+
+vec(foo, foo)
+#> [1] foo foo
+#> Levels: foo
+
+c(foo, foo)
+#> [1] 1 1
+```
+
+Warnings are consistently thrown when information is lost because of
+coercions:
+
+```
+vec(foo, "bar")
+#> [1] "foo" "bar"
+#> Warning message:
+#> Coercion of `factor` to `character` loses levels information
```

Comment:
This message doesn't feel quite right to me because it doesn't indicate why the coercion is happening. Maybe something like: "<factor> coerced to <character>, losing levels" (I like idea of standardising on `<>` for types)

Reply:
This was supposed to read

```
`<factor>` coerced to `<character>`, losing levels
```

right?